source/defclass-star.lisp

```diff
@@ -302,22 +383,30 @@ The predicate function returns true when the argument is a type of the `name' cl
                  ,@(when (or *symbols-to-export*
                              *export-class-name-p*)
                      `((eval-when (:compile-toplevel :load-toplevel :execute)
-                         (export '(,@(append (when *export-class-name-p*
-                                         (list name))
-                                       *symbols-to-export*))
+                         ;; Don't try to export symbols that don't belong to *package*.
+                         ;; This can happen when inheriting from a class and
+                         ;; overriding some slot.
+                         (export '(,@(remove-if (lambda (sym)
+                                                  (not (eq (symbol-package sym) *package*)))
+                                      (append (when *export-class-name-p*
+                                                (list name))
+                                       *symbols-to-export*)))
                                  ,(package-name *package*)))))
                  (find-class ',name nil))
-            result))))))
+              result))))))
 
-(defmacro defclass* (name supers slots &rest options)
+(defmacro define-class (name supers slots &rest options)
+  "The default types can be automatically inferred by the function specified
+in the `:type-inference' option, which defaults to `*type-inference*'.
+The type can still be specified manually with the `:type' slot option."
   (build-defclass-like-expansion
    name supers slots options
    (lambda (processed-slots clean-options)
      `(defclass ,name ,supers
         ,processed-slots
         ,@clean-options))))
 
-(defmacro defcondition* (name supers slots &rest options)
+(defmacro define-condition* (name supers slots &rest options)
```

Comment:
This feels inconsistent, maybe we can come up with a more consistent naming? Or maybe shadow the CL symbol (I shudder at the thought of it （；￣д￣）)?

Reply:
Exactly, not ideal either.  The problem is that the standard is inconsistent, so we have to live with it I guess.

Reply:
Yeah, I guess so :/

Maybe have it as in `njson`: make a separate package (`nclasses/aliases`?) with shadowed symbols, so that `nclasses` one is conflict-less, while the `nclasses/aliases` is more consistent?

Reply:
For just one symbol?  In this case, the end user would be better off shadowing the symbol themselves, I think.